crates/optimism/rpc/src/eth/ext.rs

```diff
@@ -20,37 +21,51 @@ const MAX_CONDITIONAL_EXECUTION_COST: u64 = 5000;
 /// Separate from [`super::OpEthApi`] to allow to enable it conditionally,
 #[derive(Clone)]
 #[allow(dead_code)]
-pub(crate) struct OpEthApiExt<N: OpNodeCore> {
-    /// Gateway to node's core components.
-    inner: Arc<OpEthApiInner<N>>,
+#[derive(Debug)]
+pub struct OpEthExtApi<Pool, Provider> {
+    /// Sequencer client, configured to forward submitted transactions to sequencer of given OP
+    /// network.
+    sequencer_client: Option<SequencerClient>,
+    /// The transaction pool of the node.
+    pool: Arc<Pool>,
+    /// The provider type used to interact with the node.
+    provider: Arc<Provider>,
```

Comment:
these args are redundant, we should insted group all fields into a `OpEthExtApiInner` type so that we can arc this one

Reply:
sure done here https://github.com/paradigmxyz/reth/pull/14421/commits/e94c477d9fa010e494783218db57431023461b9a